src/util/error-handler.ts

```diff
@@ -151,3 +151,132 @@ export const reactionRolesErrorHandler: ErrorHandler = async (ctx: ButtonInterac
     
     // ----- !ERROR LOG -----
 };
+
+/**
+ * Actions related to the members logs category.
+ */
+export enum MemberLogsAction {
+    MemberJoin = "Member Join",
+    MemberLeave = "Member Leave",
+    MemberBan = "Member Ban",
+    MemberUnban = "Member Unban"
+}
+
+/**
+ * Actions related to the profiles logs category.
+ */
+export enum ProfileLogsAction {
+    MemberUpdate = "Member Update",
+    UserUpdate = "User Update"
+}
+
+/**
+ * Actions related to the messages logs category.
+ */
+export enum MessageLogsAction {
+    MessageEdit = "Message Edit",
+    MessageDelete = "Message Delete"
+}
+
+/**
+ * All logs categories.
+ */
+type LogsAction = MemberLogsAction |
+    ProfileLogsAction |
+    MessageLogsAction;
+
+/**
+ * The payload of a logs error.
+ * The before and after states of a Message if it's a message edit action, or
+ * a single message if it's a message delete action, or
+ * the before and after states of a User if it's a user update action, or
+ * the before and after states of a Member if it's a member update action.
+ */
+type Payload<T extends LogsAction> = T extends MessageLogsAction.MessageEdit ? [before: Message, after: Message] :
+    T extends MessageLogsAction ? [message: Message] :
+    T extends ProfileLogsAction.UserUpdate ? [before: User, after: User] :
+    T extends ProfileLogsAction.MemberUpdate ? [before: GuildMember, after: GuildMember] :
+    [];
```

Comment:
The `payload`'s type depends on what data can be collected from the error. For message edits, 2 message objects will be collected; for user profile updates, 2 user objects will be collected; etc.